internal/pkg/gateway/api.go

```diff
@@ -432,43 +432,47 @@ func (gs *Server) Submit(ctx context.Context, request *gp.SubmitRequest) (*gp.Su
 	for _, index := range rand.Perm(len(orderers)) {
 		orderer := orderers[index]
 		logger.Infow("Sending transaction to orderer", "txID", request.TransactionId, "endpoint", orderer.address)
-		err := gs.broadcast(ctx, orderer, txn)
-		if err == nil {
+		response, err := gs.broadcast(ctx, orderer, txn)
+		if err != nil {
+			errDetails = append(errDetails, errorDetail(orderer.endpointConfig, err.Error()))
+			logger.Warnw("Error sending transaction to orderer", "txID", request.TransactionId, "endpoint", orderer.address, "err", err)
+			continue
+		}
+
+		status := response.GetStatus()
+		if status == common.Status_SUCCESS {
 			return &gp.SubmitResponse{}, nil
 		}
 
-		logger.Warnw("Error sending transaction to orderer", "txID", request.TransactionId, "endpoint", orderer.address, "err", err)
-		errDetails = append(errDetails, errorDetail(orderer.endpointConfig, err.Error()))
+		logger.Warnw("Unsuccessful response sending transaction to orderer", "txID", request.TransactionId, "endpoint", orderer.address, "status", status, "info", response.GetInfo())
+
+		logger.Warnw("Error sending transaction to orderer", "txID", request.TransactionId, "endpoint", orderer.address, "err", err, "status", status, "info", response.GetInfo())
```

Comment:
Did you intend to log two similar warnings?

Reply:
No I didn't - good spot